pkg/kubectl/cmd/apply.go

```diff
@@ -282,6 +283,12 @@ func parsePruneResources(mapper meta.RESTMapper, gvks []string) ([]pruneResource
 	return pruneResources, nil
 }
 
+func isIncompatibleServerError(err error) bool {
+	// 415: Unsupported media type means we're talking to a server which doesn't
+	// support server-side apply.
+	return err.(*errors.StatusError).Status().Code == http.StatusUnsupportedMediaType
```

Comment:
Won't this crash if the error isn't of type `*errors.StatusError`?

Reply:
The error returned is always a StatusError.

Reply:
After our discussion, I agree we should be more defensive and check if it's a StatusError first.